app/Http/Controllers/FishRecordController.php

```diff
@@ -57,8 +58,37 @@ public function delete(Request $request, $id)
         return redirect()->to('fish_records');
     }
 
-    public function detalis($id){
+    public function detalis($id)
+    {
         $rec = Fish_Record::find($id);
         return view('Fish_Record_detalis', ['rec' => $rec]);
     }
+
+    public function search()
+    {
+        $data = new Fish_Record;
+        //釣行日で絞り込み
+        $search = Request::get('fishing_date');
+        if (!empty($search)) {
+            $data = Fish_Record::whereDate('fishing_date', $search);
+        }
+        //釣行先で絞り込み
+        $search = Request::get('harbor');
+        if (!empty($search)) {
+            $data = Fish_Record::where('harbor', $search);
+        }
+        //船宿で絞り込み
+        $search = Request::get('ship');
+        if (!empty($search)) {
+            $data = Fish_Record::where('ship', $search);
+        }
+        //釣った魚で絞り込み
+        $search = Request::get('fish_name');
+        if (!empty($search)) {
+            $data = Fish_Record::where('fish_name', $search);
+        }
+        $data = $data->get();
+        return view('Fish_Records_search')->with('data', $data);
+        //外道で絞り込み
```

Comment:
これは後で追加するという意味でのコメントですかね？

もしそうであればこんな感じで書いておくとよいです
```suggestion
        // TODO: 外道で絞り込み
```